gormsupport/lifecycle_hard_delete_blackbox_test.go

```diff
@@ -0,0 +1,54 @@
+package gormsupport_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/convert"
+	"github.com/fabric8-services/fabric8-common/gormsupport"
+	"github.com/fabric8-services/fabric8-common/resource"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLifecycleHardDeleteEqual(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+
+	// Ensure LifecycleHardDelete implements the Equaler interface
+	var _ convert.Equaler = gormsupport.LifecycleHardDelete{}
+	var _ convert.Equaler = (*gormsupport.LifecycleHardDelete)(nil)
```

Comment:
this should be part of implementation rather in test code.

Reply:
like this [link](https://github.com/fabric8-services/fabric8-common/blob/master/gormsupport/lifecycle.go#L28)

Reply:
Well... I actually think it's better to keep this in test. Any particular reason to keep it lifecycle.go instead of tests?

Reply:
I will add it under impl as `LifecycleHardDelete.Equal()` is part of impl .. also this is optional so adding it under test also would be fine.